Sources/BowFree/Coyoneda.swift

```diff
@@ -2,66 +2,157 @@ import Foundation
 import Bow
 
 public final class ForCoyoneda {}
-public typealias AnyFunc = (AnyObject) -> AnyObject
-public final class CoyonedaPartial<F, P>: Kind2<ForCoyoneda, F, P> {}
-public typealias CoyonedaOf<F, P, A> = Kind<CoyonedaPartial<F, P>, A>
+public final class CoyonedaPartial<F>: Kind<ForCoyoneda, F> {}
+public typealias CoyonedaOf<F, A> = Kind<CoyonedaPartial<F>, A>
 
-public class Coyoneda<F, P, A>: CoyonedaOf<F, P, A> {
-    fileprivate let pivot: Kind<F, P>
-    fileprivate let ks: [AnyFunc]
-
-    public static func apply(_ fp : Kind<F, P>, _ f : @escaping (P) -> A) -> Coyoneda<F, P, A> {
-        return unsafeApply(fp, [f as! AnyFunc])
-    }
-
-    public static func unsafeApply(_ fp : Kind<F, P>, _ fs : [AnyFunc]) -> Coyoneda<F, P, A> {
-        return Coyoneda<F, P, A>(fp, fs)
+public class Coyoneda<F, A>: CoyonedaOf<F, A> {
+    internal let pivot: Kind<F, /*B*/Any>
+    internal let function: (/*B*/Any) -> A
+    
+    public init(
+        _ pivot: Kind<F, Any>,
+        _ function: @escaping (Any) -> A
+    ) {
+        self.pivot = pivot
+        self.function = function
     }
-
-    public static func fix(_ fa : CoyonedaOf<F, P, A>) -> Coyoneda<F, P, A> {
-        return fa as! Coyoneda<F, P, A>
+    
+    public static func fix(_ fa: CoyonedaOf<F, A>) -> Coyoneda<F, A> {
+        fa as! Coyoneda<F, A>
     }
+}
 
-    public init(_ pivot : Kind<F, P>, _ ks : [AnyFunc]) {
-        self.pivot = pivot
-        self.ks = ks
+public extension Coyoneda where F: Functor {
+    static func liftCoyoneda(_ fa: Kind<F, A>) -> Coyoneda<F, A> {
```

Comment:
In Haskell [`liftCoyoneda`](https://hackage.haskell.org/package/kan-extensions-5.2/docs/Data-Functor-Coyoneda.html#v:liftCoyoneda) does not require F to be a functor. Can we achieve this here too?

Reply:
I'm trying to port Haskell's Program (Free-er monad), and I need this here: http://hackage.haskell.org/package/reasonable-operational-0.1.0.1/docs/src/Control-Monad-Operational.html#singleton

Reply:
I've not been able to remove this. The problem is that I cannot go from `Kind<F, A>` to `Kind<F, Any>` without `map`ing to do the cast. Any idea how we can achieve this?